### PR TITLE
Tabular: Added custom metric tutorial

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Added custom metric tutorial
- Sped up regression metric calculations by ~50%
- Removed error_bad_lines in load_pd as it is deprecated in pandas 1.3

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1271-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1271/1/index.html

**Comment by szha** [comment]
Job PR-1271-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1271/2/index.html

**Comment by szha** [comment]
Job PR-1271-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1271/3/index.html

**Comment by Innixma** [comment]
Tutorial Page: http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1271/3/tutorials/tabular_prediction/tabular-custom-metric.html

File: core/src/autogluon/core/metrics/__init__.py [hunk 1]
```diff
@@ -104,17 +104,20 @@ def __call__(self, y_true, y_pred, sample_weight=None):
...
+        if len(y_pred.shape) == 1 or y_pred.shape[1] == 1:
             pass  # must be regression, all other task types would return at least two probabilities
-        elif type_true in ['binary', 'multiclass']:
-            y_pred = np.argmax(y_pred, axis=1)
-        elif type_true == 'multilabel-indicator':
-            y_pred[y_pred > 0.5] = 1.0
-            y_pred[y_pred <= 0.5] = 0.0
         else:
-            raise ValueError(type_true)
+            type_true = type_of_target(y_true)
```
Comment (Innixma) [inline core/src/autogluon/core/metrics/__init__.py:111]: This change avoids computing type_true until necessary, speeds up regression metric calculation by ~50% in a lot of cases.

**Comment by szha** [comment]
Job PR-1271-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1271/4/index.html

**Review by sxjscience — APPROVED** [review]

**Pull request merged**